### DISCUSSION TITLE
Cursor and highlighting broken when adjusting frame

### DISCUSSION THREAD

**Discussion by sebjvidal** [krzyzanowskim/STTextView#36] [Questions]
The cursor and highlighting view seem to be offset by the textView's distance from (0,0). For example, setting the textView's `frame.origin` to `CGPoint(x: 10, y: 10)` results in the following cursor and highlighting view offset:
|Window|Xcode Debug View Hierarchy|
|-|-|
|<img width="527" alt="Screenshot 2023-10-05 at 12 28 54" src="https://github.com/krzyzanowskim/STTextView/assets/27621026/601c9a97-2568-43a5-8591-df337853ed1d">|![Screenshot 2023-10-05 at 12 33 18](https://github.com/krzyzanowskim/STTextView/assets/27621026/ade9c96f-6cf5-4a86-8f04-4a2f72f2af38)|

This can be reproduced with the following code on macOS 13.5.2 using Xcode 15.0 and STTextView 0.8.15:

```swift
import AppKit
import STTextView

class ViewController: NSViewController, STTextViewDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()

        let textView = STTextView()
        textView.string = "Snippet"
        textView.backgroundColor = .systemTeal
        textView.font = .monospacedSystemFont(ofSize: 15, weight: .regular)
        textView.frame.origin = CGPoint(x: 0, y: 0) // Set `frame.origin` of `textView`
        
        view.addSubview(textView)
    }
}
```

This can cause the cursor and highlighting view to disappear completely with more drastic offsets, for example centering the textView in its superview:

https://github.com/krzyzanowskim/STTextView/assets/27621026/61bf4cd9-9fc7-4c6a-a4ee-28982767fc14

**Comment by krzyzanowskim** [comment]
just checking, do you still can reproduce that issue?
